frame/dutch-auction/src/lib.rs

```diff
@@ -110,204 +57,64 @@ pub mod pallet {
-			OrdersIndex::<T>::set(order_id.next());
-
-			let order = Order {
-				started: T::UnixTime::now().as_secs(),
-				function,
-				account_id: account_id.clone(),
-				source_asset_id,
-				source_account: source_account.clone(),
-				target_asset_id,
-				target_account: target_account.clone(),
-				source_total_amount: total_amount,
-				source_initial_price: price,
-				state: AuctionState::AuctionStarted,
-			};
-			Orders::<T>::insert(order_id.clone(), order);
-			Self::deposit_event(Event::<T>::AuctionWasStarted { order_id: order_id.clone() });
-			Ok(order_id)
+			Self::deposit_event(Event::OrderAdded { order_id: <_>::default() });
+			todo!()
 		}
 
-		fn off_chain_run_auctions(now: Timestamp) -> DispatchResult {
-			// avoid removing during iteration as unsafe
-			let mut removed = Vec::new();
-			for (order_id, order) in Orders::<T>::iter() {
-				match order.state {
-					AuctionState::AuctionStarted => {
-						if now > order.started + ONE_HOUR {
-							removed.push(order_id);
-						} else {
-							// for final protocol may be will need to transfer currency onto auction
-							// pallet sub account and send dex order with idempotency tracking id final protocol seems should include multistage lock/unlock https://github.com/paritytech/xcm-format or something
-							let delta_time = now - order.started;
-							let price: LiftedFixedBalance =
-								order.source_initial_price.initial_price.into();
-							let total_price = price.safe_mul(&order.source_total_amount.into())?;
-							let price = match order.function {
-								AuctionStepFunction::LinearDecrease(parameters) =>
-									parameters.price(total_price, delta_time),
-								AuctionStepFunction::StairstepExponentialDecrease(parameters) =>
-									parameters.price(total_price, delta_time),
-							}?
-							.checked_mul_int(1u64)
-							.ok_or(ArithmeticError::Overflow)?;
-
-							let price =
-								Price::<Self::GroupId, Self::Balance>::new_any(price.into());
-
-							let dex_order_intention = <T::Orderbook as Orderbook>::post(
-								&order.account_id,
-								order.source_asset_id,
-								order.target_asset_id,
-								order.source_total_amount,
-								price,
-								Permill::from_perthousand(5),
-							)?;
-
-							Orders::<T>::mutate(order_id, |order| {
-								// considers updating in place is
-								// safe during iteration
-								order.state = AuctionState::AuctionOnDex(dex_order_intention.id);
-							});
-						}
-					},
-					AuctionState::AuctionOnDex(_) => {
-						// waiting for off chain callback about order status
-						if now > order.started + ONE_HOUR {
-							removed.push(order_id);
-						}
-					},
-					_ => {
-						removed.push(order_id);
-					},
-				}
-			}
-
-			for r in removed.iter() {
-				Orders::<T>::remove(r);
-			}
-
-			Ok(())
+		fn take(
+			_from_to: &Self::AccountId,
+			_order: Self::OrderId,
+			_take: Take<Self::Balance>,
+		) -> Result<(), DispatchError> {
+			todo!()
```

Comment:
Next PR as well for full implementation?

Reply:
yes. i have some non linted implementation.